src/internal.c

```diff
@@ -10534,6 +10562,194 @@ static void FreeProcPeerCertArgs(WOLFSSL* ssl, void* pArgs)
+                suffix = ph->last_suffix;
+            } else {
+                ph = NULL;
+                suffix = 0;
+            }
+            
+            wc_UnLockMutex(&lookup->dirs->lock);
+        }
+        
+        /* Additional buffer length for file name memory allocation :   */
+        /* / <hashvalue>.(r)N\0                                         */
+        /*|1|     8    |1|1|1|1|           => 13                        */
+        len = (int)XSTRLEN(entry->dir_name) + 13;
+        if (filename != NULL) {
+            XFREE(filename, NULL, DYNAMIC_TYPE_OPENSSL);
+        }
+        
+        filename = (char*)XMALLOC(len, NULL, DYNAMIC_TYPE_OPENSSL);
+        if (filename == NULL) {
+            WOLFSSL_MSG("memory allocation error");
+            return MEMORY_E;
+        }
+        
+        /* set as FAILURE, if successfuly loading cert of CRL, this becomes */
+        /* WOLFSSL_SUCCESS                                                  */
+        ret = WOLFSSL_FAILURE;
+        
+        for (; suffix < MAX_SUFFIX; suffix++) {
+            /* /folder-path/<hash>.(r)N[0..9] */
+            XSNPRINTF(filename, len, "%s/%08lx.%s%d", entry->dir_name, 
+                                                       hash, post, suffix);
+            if(wc_FileExists(filename) == 0/*0 file exists */) {
+                
+                if (type == X509_LU_X509) {
+                    x509 = wolfSSL_X509_load_certificate_file(filename, 
+                                                        WOLFSSL_FILETYPE_PEM);
+                    if (x509 != NULL) {
+                       ret = wolfSSL_X509_STORE_add_cert(store, x509);
+                       wolfSSL_X509_free(x509);
+                    } else {
+                       WOLFSSL_MSG("failed to load certificate\n");
+                       ret = WOLFSSL_FAILURE;
+                       break;
+                    }
+                } 
+#ifdef HAVE_CRL
+                else if (type == X509_LU_CRL) {
+                    ret = wolfSSL_X509_load_crl_file(&store->lookup, filename,
+                                                    WOLFSSL_FILETYPE_PEM);
+                    if (ret != WOLFSSL_SUCCESS) {
+                        WOLFSSL_MSG("failed to load CRL\n");
+                        break;
+                    }
+                }
+#else
+                else if (type == X509_LU_CRL) {
+                    WOLFSSL_MSG("CRL is not supported");
+                    ret = WOLFSSL_FAILURE;
+                    break;
+                }
+#endif
+            } else
+                break;
+        }
+        
+        if (ret != WOLFSSL_SUCCESS) {
+            WOLFSSL_MSG("not found file");
+            ret = WOLFSSL_FAILURE;
+        } else {
+            if (type == X509_LU_CRL) {
+                if (wc_LockMutex(&lookup->dirs->lock) != 0) {
+                    WOLFSSL_MSG("wc_LockMutex cdir Lock error");
+                    return BAD_MUTEX_E;
```

Comment:
Looks like we need to XFREE filename here before returning early.

Reply:
Thank you for pointing out.